### PR TITLE
Added 3 methods to add, delete, and redefine job search terms

### PR DESCRIPTION
**Pull request by AustinJamesWolff**
## Description

While the job description scraper works, someone pointed out it would be nice to be able to change what positions the scraper looks for (again, this package is for purposes of gathering job description data for the ML Engineers to build an NLP model to match mentee's resumes to similar job openings). The three methods involve adding job titles, removing job titles, and redefining the job title search list.

Loom: https://www.loom.com/share/0a9192e12e074d9eb462f76577389522

Fixes # (issue)

- [x] New feature

## Checklist:

- [x] My code follows PEP8 style guide
- [x] I have removed unnecessary print statements from my code
- [x] I have made corresponding changes to the documentation if necessary
- [x] My changes generate no errors
- [x] No commented-out code
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by DagimB — APPROVED** [review]
Very forward-thinking/user-friendly and very organized in code structure. Wonderful job!

File: data_generators/job_description_scraper.py [hunk 1]
```diff
@@ -564,6 +564,89 @@ def combine_individual_datasets(self, folder_date):
...
+    def add_job_titles(self, job_list):
+        """
+        Add more job titles to search through.
+
+        Args:
+            job_list (list): A list of job titles in HTML
+            format for the scraper to search through.
+            Remember that empty spaces in words need to
+            be replaced with "%20" to stay HTML compliant.
+                ex: ["web%20developer","backend%20engineer"]
```
Comment (BrokenShell) [inline data_generators/job_description_scraper.py:578]: Can the space conversion to `%20` be automated so our users don't need to remember this? Currently it breaks encapsulation in the sense that our http requirements are leaking. Look for something called url encoding, I'm pretty sure Python has a prebuilt solution for this in the standard library.

File: data_generators/job_description_scraper.py [hunk 2]
```diff
@@ -564,6 +564,89 @@ def combine_individual_datasets(self, folder_date):
...
+                ex: ["devops"]
+
+        Returns:
+            None.
+        """
+        # Check if a list was passed
+        if type(job_list) == list:
```
Comment (BrokenShell) [inline data_generators/job_description_scraper.py:638]: Checking a type like this isn't the best way to go. If it's actually required it's better to use the is_instance built in function. Also it's better to compare with the generics and not specific types, ie. Sequence for list-like objects. `isinstance(["some list"], Sequence)`. Moreover this type of check is not helping the code very much, it's mostly constricting what our users can do and not providing much benefit. I recommend trying set operations for finding the items in one sequence that aren't in another - just an idea.

File: data_generators/job_description_scraper.py [hunk 3]
```diff
@@ -564,6 +564,89 @@ def combine_individual_datasets(self, folder_date):
...
+            # Create new list without the search terms you
+            # want to remove using list comprehension
+            self.search_terms = [
+                x for x in self.search_terms if x not in job_list
+            ]
+            print("Success.")
+        else:
+            print("Unsuccessful. Please pass a list.")
+
```
Comment (BrokenShell) [inline data_generators/job_description_scraper.py:647]: Return values would be better than embedded print statements.

File: data_generators/job_description_scraper.py [hunk 4]
```diff
@@ -564,6 +564,89 @@ def combine_individual_datasets(self, folder_date):
...
+            Remember that empty spaces in words need to
+            be replaced with "%20" to stay HTML compliant.
+                ex: ["web%20developer","backend%20engineer"]
+                ex: ["devops"]
+
+        Returns:
+            None.
+        """
+        # Check if a list was passed
+        if type(job_list) == list:
```
Comment (BrokenShell) [inline data_generators/job_description_scraper.py:611]: See below.

File: data_generators/job_description_scraper.py [hunk 5]
```diff
@@ -564,6 +564,89 @@ def combine_individual_datasets(self, folder_date):
...
```
Comment (BrokenShell) [inline data_generators/job_description_scraper.py:585]: See below.

**Review by BrokenShell — CHANGES_REQUESTED** [review]
Great job over all. You might want to consider the possibility that these three methods could be refactored into just one.

**Comment by BrokenShell** [comment]
> Austin - This is really a great "mini application".
> 
> I don't want to clutter up what you've built or take this in a different direction, but for lazy people that don't want to add their LinkedIn profiles to their environment variables, could we update the credentials section to something like this:
> 
> ```
>     # Save LinkedIn username and password to variables
>     try:
>         linkedin_user = os.environ['LINKEDIN_USER']
>     except:
>         linkedin_user = raw('Enter your LinkedIn user id: ')
> 
>     try:
>         linkedin_pass = os.environ['LINKEDIN_PASS']
>     except:
>         linkedin_pass = raw('Enter your LinkedIn password: ')
> ```
> 
> This would lower the 'barrier to entry' for anyone testing the feature, running it on a remote machine, etc.

`raw(...)`? Did you mean `input(...)`

**Comment by hollberg** [comment]
> 

Yes - should be 'input' and not 'raw'.

File: data_generators/job_description_scraper.py (hunk 3 above)
Comment (AustinJamesWolff) [reply data_generators/job_description_scraper.py:647]: Done!

File: data_generators/job_description_scraper.py (hunk 2 above)
Comment (AustinJamesWolff) [reply data_generators/job_description_scraper.py:638]: Done!

File: data_generators/job_description_scraper.py (hunk 1 above)
Comment (AustinJamesWolff) [reply data_generators/job_description_scraper.py:578]: Done!

**Review by BrokenShell — APPROVED** [review]

**Review by Zack-Quintana — APPROVED** [review]
The only things that I can see that needs to be addressed already were. Seems good to me!

**Pull request merged**